AutomationHelpers/src/UserCodeCollections/FileLibrary.cs

```diff
@@ -152,9 +153,169 @@ public static void WaitForFile(string path, string pattern, int duration, int in
+                            Report.Failure("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");
+                            return;
+                        }
+                    }
+                }
+
+                Report.Success("Files '" + filePath1 + "' and '" + filePath2 + "' are equal.");
+            }
+            catch (Exception ex)
+            {
+                Report.Error(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Compares content of two text files.
+        /// </summary>
+        /// <param name="filePath1">The relative or absolute path of the first file</param>
+        /// <param name="filePath2">The relative or absolute path of the second file</param>
+        [UserCodeMethod]
+        public static void CompareTextFiles(string filePath1, string filePath2)
+        {
+            try
+            {
+                filePath1 = GetPathForFile(filePath1);
+                filePath2 = GetPathForFile(filePath2);
+
+                if (!FilesExist(filePath1, filePath2))
+                {
+                    return;
+                }
+
+                var fileContent1 = File.ReadAllText(filePath1);
+                fileContent1 = Regex.Replace(fileContent1, "(\r\n)|(\n\r)|(\n)|(\r)", "\r\n");
+
+                var fileContent2 = File.ReadAllText(filePath2);
+                fileContent2 = Regex.Replace(fileContent2, "(\r\n)|(\n\r)|(\n)|(\r)", "\r\n");
+
+                if (fileContent1 != fileContent2)
+                {
+                    Report.Failure("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");
+                    return;
+                }
+
+                Report.Success("Files '" + filePath1 + "' and '" + filePath2 + "' are equal.");
+            }
+            catch (Exception ex)
+            {
+                Report.Error(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Checks if file contains text specified.
+        /// </summary>
+        /// <param name="filePath">The relative or absolute path to the file</param>
+        /// <param name="text">The text to search for</param>
+        [UserCodeMethod]
+        public static void SearchText(string filePath, string text)
+        {
+            try
+            {
+                filePath = GetPathForFile(filePath);
+
+                if (!FilesExist(filePath))
+                {
+                    return;
+                }
+
+                var textFound = false;
+
+                using (StreamReader sr = new StreamReader(filePath))
+                {
+                    var line = "";
+                    var i = 1;
+
+                    while ((line = sr.ReadLine()) != null)
+                    {
+                        if (line.IndexOf(text, StringComparison.OrdinalIgnoreCase) != -1)
+                        {
+                            Report.Success("The text '" + text + "' was found on the line " + i + ": '" + line + "'.");

```

Comment:
Remove the "the".
Nice idea to mention the line on success!

Reply:
Done